src/vs/workbench/services/search/electron-browser/searchService.ts

```diff
@@ -84,6 +86,13 @@ export class DiskSearch implements ISearchResultProvider {
 		const client = new Client(FileAccess.asFileUri('bootstrap-fork', require).fsPath, opts);
 		const channel = getNextTickChannel(client.getChannel('search'));
 		this.raw = new SearchChannelClient(channel);
+
+		const disposeClient = async () => {
+			client.dispose();
+		};
+		this.lifecycleService.onWillShutdown((e) =>
```

Comment:
If the disposal is not `async` you don`t have to do this joining on the event, just call the `dispose` from the event without `async` and without joining.